src/Microsoft.AspNet.Mvc.Core/ModelBinding/Metadata/DefaultModelMetadataProvider.cs

```diff
@@ -122,7 +122,8 @@ protected virtual DefaultMetadataDetails[] CreatePropertyDetails([NotNull] Model
                 var propertyEntry = new DefaultMetadataDetails(propertyKey, attributes);
                 if (propertyHelper.Property.CanRead && propertyHelper.Property.GetMethod?.IsPublic == true)
                 {
-                    propertyEntry.PropertyGetter = propertyHelper.ValueGetter;
+                    var getter = PropertyHelper.MakeNullSafeFastPropertyGetter(propertyHelper.Property);
+                    propertyEntry.PropertyGetter = getter;
```

Comment:
Should test `null` safety in `DefaultModelMetadataProviderTest`. At the moment nothing except product code (`MutableObjectModelBinder` i.e. no test code) does anything beyond check that `metadata.PropertyGetter` is itself non-`null`.